File: API/ChatAPI/Core/DTOs/MessageDTO.cs
namespace Core.DTOs
{
    public class MessageDTO
    {
        public int Id { get; set; }
        public string Text { get; set; }
        public UserDTOToUI User { get; set; }
        public MessageDTO? Replied { get; set; }

        public int GroupId { get; set; }

        public DateTime Created { get; set; }
    }
}

File: UI/chat-ui/src/app/models/messageDTO.ts
import { BaseEntity } from "./base-entities/base-entity";
import { UserDTOFromServer } from "./userDTO-from-server";

export interface MessageDTO extends BaseEntity{
    groupId: number;
    text: string;
    user: UserDTOFromServer;
    replied?: MessageDTO;
    created: Date;
}